lib/toolbox/saved_struct.c

```diff
@@ -0,0 +1,149 @@
+#include "saved_struct.h"
+#include <furi.h>
+#include <stdint.h>
+#include <storage/storage.h>
+
+#define SAVED_STRUCT_TAG    "SAVED_STRUCT"
+
+typedef struct {
+    uint8_t magic;
+    uint8_t version;
+    uint8_t checksum;
+    uint8_t flags;
```

Comment:
Also I don't remember why I've made flags field, but it was something specific to DolphinState itself. Probably make sense to drop it and give this byte to magic.